src/write.jl

```diff
@@ -26,17 +26,45 @@ end
 Write out a `.pomdp` file using the POMDPs.jl interface
 Specification: http://cs.brown.edu/research/ai/pomdp/examples/pomdp-file-spec.html
 """
-function Base.write(io::IO, pomdp::POMDP)
+function Base.write(io::IO, pomdp::POMDP; pretty = false)
 
     ss = states(pomdp)
     as = actions(pomdp)
     os = observations(pomdp)
 	println(io, "discount: ", discount(pomdp))
 	println(io, "values: reward") # NOTE(tim): POMDPs.jl assumes rewards rather than costs
-    println(io, "states: ", length(ss))
-    println(io, "actions: ", length(as))
-    println(io, "observations: ", length(os), "\n")
-	
+
+	n_states = length(ss)
+	if pretty && n_states < 20
+    	println(io, "states: ", join([string(s) for s=ss], " "))
```

Comment:
We should not use `string` for this. `string` will produce things like `State([1, 2])` which will be extremely difficult for a parser to handle. I think we should by default replace all non-alphanumeric characters with underscores and allow the user to override this.

Reply:
I did `string(s)` to allow users to implement their own `string` method for the state. I'm not super familiar with Julia, so it's possible another string-like method would be suitable, but I figure that "pretty printing" does have some overhead on the user's end for what constitute's "readable" names.

Reply:
The downside of having them specialize `string` for this specific use is that `string` will behave that way everywhere else in julia. It may well be that the easiest-to-read representation of a state has spaces in it. But in this context, the representation *cannot* have spaces. So if they override string just for this purpose it will muck up everything else.

There are two alternatives
1. Have a function like `POMDPFiles.state_string` that the user can specialize for their type if they want.
2. Just let the user use any function that they pass in as an argument.

I prefer 2 because it is more flexible and easier to change later if they want. It would look something like this:
```julia
function pretty_print(io::IO, m::POMDP; name::Function=x -> non_alphanumeric_to_underscore(string(x)))
    # ...
    println(io, "states: ", join([name(s) for s=ss], " "))
    # ...
end
```
then you write `non_alphanumeric_to_underscore` using regexes or something and put it in this package. Then there is a sensible default, and if the user wants to override it, they can do
```julia
pretty_print(io, m, name = s -> "$(s.x)_$(s.y)")
```
or whatever they want.

Reply:
I went the route of individual functions for `states`, `actions`, and `observations`.